beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -201,4 +202,20 @@ public BlockOperationSelectorFactory(
                 })
             .join();
   }
+
+  public Consumer<BeaconBlockUnblinder> createUnblinderSelector() {
+    return bodyUnblinder -> {
+      if (isMevBoostEnabled) {
+        bodyUnblinder.executionPayload(
+            () ->
+                executionEngineChannel
+                    .proposeBlindedBlock(bodyUnblinder.getSignedBlindedBeaconBlock())
+                    .join());
```

Comment:
It would be really nice to design this to not use `.join()` here and return a `SafeFuture`. At some point in the block factory stuff we may need a `.join()` because block creation is currently synchronous but the higher up the call stack we can push it the better as it will be easier to get rid of later then.

Reply:
yeah, good exercise for the complete removal.